### PR TITLE
persist filter option across sessions

### PR DESCRIPTION
**Pull request by 0rp3u**
Persists the filter options across sessions.
Requesting @miguelpruivo review, since he closed the issue for creating the filters.

### CODE REVIEW

File: lib/middleware/preferences_middleware.dart [hunk 1]
```diff
@@ -40,7 +43,13 @@ Middleware<AppState> _createLoadPreferences() {
       }
 
       List<String> subbedFires = prefs.getStringList('subscribedFires') ?? [];
-      List<Fire> fires = store.state.fires;
+      List<Fire> fires = List.from(store.state.fires);
```
Comment (miguelpruivo) [inline lib/middleware/preferences_middleware.dart]: Why are you forcing a new reference here?

Comment (0rp3u) [reply lib/middleware/preferences_middleware.dart]: I actually wanted to ask about that, but a day passed and i forgot to, if i don't i get a` type 'List<dynamic>' is not a subtype of type 'List<Fire>'`.
Any idea?

Comment (miguelpruivo) [reply lib/middleware/preferences_middleware.dart]: It should implicitly cast to `List<Fire>` without problem in runtime. What you're doing is redundant anyway. 

Did you have any issue without it? Just ran through it with `List<Fire> fires = store.state.fires;` and it was good.

Comment (miguelpruivo) [reply lib/middleware/preferences_middleware.dart]: It should implicitly cast to `List<Fire>` without problem in runtime. What you're doing is redundant anyway.

Did you have any issue without it? Just ran through it with `List<Fire> fires = store.state.fires;` and it was looking good.

Comment (miguelpruivo) [reply lib/middleware/preferences_middleware.dart]: It should implicitly cast to `List<Fire>` without problem at runtime. What you're doing is redundant anyway.

Did you have any issue without it? Just ran through it with `List<Fire> fires = store.state.fires;` and it was looking good.

Comment (0rp3u) [reply lib/middleware/preferences_middleware.dart]: Yes, it throws a  `type 'List<dynamic>' is not a subtype of type 'List<Fire>'` on that line, and the filter is only applied after i refresh the fires or change the filter.
Could you run it in debug mode and verify that you get the same exception on the catch clause?

Comment (miguelpruivo) [reply lib/middleware/preferences_middleware.dart]: I've already tried and didn't seem to throw anything. I've even added a breakpoint to the line below and the fires were loading great.

What are the steps to reproduce this?

File: lib/models/fire.dart [hunk 2]
```diff
@@ -102,7 +102,7 @@ class Fire {
     );
   }
 
-  static FireStatus _statusFromJson(String status) {
```
Comment (guivazcabral) [inline lib/models/fire.dart]: why did you change `_statusFromJson` to `statueFromJson` ?

**Comment by guivazcabral** [comment]
Waiting for @miguelpruivo question to be answered and then for his review. I'd just like to know the reason for that function name change as stated in my comment.

**Comment by miguelpruivo** [comment]
@guivazcabral @Orp3eu and I have been in touch in a Discord private chat about this. 👍🏻

File: lib/models/fire.dart (hunk 2 above)
Comment (miguelpruivo) [reply lib/models/fire.dart]: Solved in `efbbc7e`.

**Comment by miguelpruivo** [comment]
You're good to go @guivazcabral. 👍

File: lib/middleware/preferences_middleware.dart (hunk 1 above)
Comment (miguelpruivo) [reply lib/middleware/preferences_middleware.dart]: Solved in [
55155a6](https://github.com/FogosPT/fogosmobile/pull/74/commits/55155a673360153de2e612d828d0c11e8cb931e9)
